updating requirements

SalahedeenIssa opened the pull request.

Commit:
updating requirements

Commit:
updating publish, release, changelog, dev, version #

Commit:
nit

rbrazinskas commented on setup.cfg:
```diff
@@ -1,6 +1,6 @@
 [metadata]
 name = qiskit-qir
```
Should the name be changed to `iqm-qiskit-qir`?

```suggestion
name = iqm-qiskit-qir
```

rbrazinskas commented on CHANGELOG.md:
```diff
@@ -0,0 +1,5 @@
+# Version 0.6.0
+
+- First version forked from the PyQIR repository. Includes support for the RGate that is part of IQM's native gateset.
```
```suggestion
- First version forked from the `qiskit-qir` repository. Includes support for the RGate that is part of IQM's native gateset.
```

rbrazinskas commented on .github/workflows/publish.yml:
```diff
@@ -0,0 +1,75 @@
+        run: |
+          ./docbuild
+          touch build/sphinx/html/.nojekyll  # allow underscores in URL path
+          # remove unnecessary build files
+          sudo rm -rf build/sphinx/html/.doctrees
```
Should we add the step to remove the `.doctrees` folder in `iqm-pyqir` "doc publish" workflow?

I think this folder has some binary files that are not needed and they might grow the size of the repository unnecessarily over time.

rbrazinskas committed:
Update Python package and CI configuration

- Test package with Python 3.12
- Set upper limit of supported Python versions to 3.12
- Trigger test workflow for pull requests against main branch

SalahedeenIssa committed:
Update setup.cfg

Co-authored-by: Ricardas Brazinskas <103200516+rbrazinskas@users.noreply.github.com>

SalahedeenIssa committed:
Update CHANGELOG.md

Co-authored-by: Ricardas Brazinskas <103200516+rbrazinskas@users.noreply.github.com>
